react/components/CreditCard.tsx

```diff
@@ -0,0 +1,303 @@
+const iframeURLDev = `https://checkoutio.vtexlocal.com.br:${PORT}/`
+
+const { production, query } = __RUNTIME__
+
+const LOCAL_IFRAME_DEVELOPMENT =
+  !production && query.__localCardUi !== undefined
+
+const iframeURL = LOCAL_IFRAME_DEVELOPMENT ? iframeURLDev : iframeURLProd
+
+const getPaymentData = (
+  {
+    encryptedCardHolder,
+    encryptedCardNumber,
+    encryptedCsc,
+    encryptedExpiryDate,
+  }: EncryptedCard,
+  selectedPaymentSystem: PaymentSystem,
+  document: string
+) => {
+  return {
+    paymentSystem: selectedPaymentSystem.name,
+    cardHolder: encryptedCardHolder,
+    cardNumber: encryptedCardNumber,
+    csc: encryptedCsc,
+    expiryDate: encryptedExpiryDate,
+    document,
+    documentType: 'CPF',
+    partnerId: '12345',
+    address: {
+      postalCode: '22775-130',
+      street: 'Rua Jaime Poggi',
+      neighborhood: 'Jacarepagua',
+      city: 'Rio de Janeiro',
+      state: 'RJ',
+      country: 'BRA',
+      number: '5001',
+      complement: 'Apto 007',
+    },
+  }
+}
+
+const CreditCard: React.FC = () => {
+  const {
+    orderForm: {
+      paymentData: { paymentSystems },
+    },
+  } = useOrderForm()
+  const [iframeLoading, setIframeLoading] = useState(true)
+
+  const [
+    selectedPaymentSystem,
+    setSelectedPaymentSystem,
+  ] = useState<PaymentSystem | null>(null)
+
+  const [doc, setDoc] = useState<Field>({
+    value: '',
+    error: false,
+    errorMessage: '',
+    showError: false,
+  })
+
+  const [saveLoading, setSaveLoading] = useState(false)
+  const [savedCard, setSavedCard] = useState<any>(null)
+
+  const iframeRef = useRef<HTMLIFrameElement>(null)
+  const {
+    culture: { locale },
+  } = useRuntime()
+  const isSSR = useSSR()
+  const intl = useIntl()
+  const { savePaymentData } = useOrderPayment()
+
+  const creditCardPaymentSystems = useMemo(
+    () =>
+      paymentSystems.filter(
+        (paymentSystem: PaymentSystem) =>
+          paymentSystem.groupName === 'creditCardPaymentGroup'
+      ),
+    [paymentSystems]
+  )
+
+  const setupIframe = useCallback(async () => {
+    iFrameResize(
+      {
+        heightCalculationMethod: 'max',
+        checkOrigin: false,
+        resizeFrom: 'parent',
+        autoResize: true,
+      },
+      iframeRef.current
+    )
+
+    const stylesheetsUrls = Array.from(
+      document.head.querySelectorAll<HTMLLinkElement>('link[rel=stylesheet]')
+    ).map(link => link.href)
+
+    await postRobot.send(iframeRef.current!.contentWindow, 'setup', {
+      stylesheetsUrls,
+      paymentSystems: creditCardPaymentSystems,
+    })
+    setIframeLoading(false)
+  }, [creditCardPaymentSystems])
+
+  const getEncryptedCard = useCallback(async (): Promise<EncryptedCard | null> => {
+    const { data } = await postRobot.send(
+      iframeRef.current!.contentWindow,
+      'encryptedCard'
+    )
+    return data
+  }, [])
+
+  useEffect(function createPaymenteSystemListener() {
```

Comment:
just found this typo

```suggestion
  useEffect(function createPaymentSystemListener() {
```